Zotero/Scenes/AppCoordinator.swift

```diff
@@ -364,30 +364,7 @@ extension AppCoordinator: AppDelegateCoordinatorDelegate {
                 completion()
                 return
             }
-
-            let disposeBag = DisposeBag()
-
-            downloader.observable
-                .observe(on: MainScheduler.instance)
-                .subscribe(onNext: { [weak self] update in
-                    guard let self, update.libraryId == attachment.libraryId && update.key == attachment.key else { return }
-
-                    switch update.kind {
-                    case .ready:
-                        completion()
-                        downloadDisposeBag = nil
-
-                    case .cancelled, .failed:
-                        downloadDisposeBag = nil
-
-                    case .progress:
-                        break
-                    }
-                })
-                .disposed(by: disposeBag)
-
-            downloadDisposeBag = disposeBag
-            downloader.downloadIfNeeded(attachment: attachment, parentKey: parentKey)
+            downloader.downloadIfNeeded(attachment: attachment, parentKey: parentKey, completion: { _ in completion() })
```

Comment:
In case of an error before, `completion` was not called, but probably it doesn't affect anything. We could consider passing the result there, if necessary.

Reply:
I know and it felt more like a bug before, so I added completion even in case of error.